sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -417,6 +417,7 @@ def fetch_metric(self, rows, columns):
         file_id = columns.index("file_id")
         file_type = columns.index("type")
         file_location = columns.index("physical_name")
+        db_files_state_desc_index = columns.index("state_desc")
```

Comment:
This is not related to AlwaysOn metrics, but rather just to add the DB state description in addition to the DB state metric. Can open a new PR for this small change to be more clear.